components/export_table/ExportTableButton.vue

```diff
@@ -1,8 +1,98 @@
 <script setup lang="ts">
+import type {Teacher} from "~/model/teacher"
+import type {DropdownItem} from "#ui/types"
+import {useTableExportStore} from "~/stores/tableExportStore"
+
+
+const tableExportStore = useTableExportStore()
+const supportedFileFormate: string[] = [
+  "CSV",
+  "JSON",
+]
+
+let selectedItem: DropdownItem = {label: ""}
+const items: DropdownItem[][] = []
+
+supportedFileFormate.forEach((format, index) => {
+  items.push([
+    {
+      label: format,
+      shortcuts: ["" + (index + 1)],
+      disabled: (index == 0),
+      icon: (index == 0) ? "i-heroicons-check-16-solid" : "",
+      class: "opacity-100",
+      click: () => selectFormat(index),
+    },
+  ])
+  if (index == 0) selectedItem = items[index][0]
+})
+
+function download() {
+  const teachers: Teacher[] = tableExportStore.teachers
+  const file = createFile(teachers)
+
+  if (file != null) {
+    const link = document.createElement("a")
+    const url = URL.createObjectURL(file)
+
+    link.href = url
+    link.download = file.name
+    document.body.appendChild(link)
+    link.click()
+
+    document.body.removeChild(link)
+    window.URL.revokeObjectURL(url)
+  }
+}
+
+function createFile(teachers: Teacher[]): File | null {
+  if (selectedItem.label == "CSV") {
+    return new File([jsonToCsv(teachers)], "untisPlanner.csv", {
+      type: "text/csv",
+    })
+  } else if (selectedItem.label == "JSON") {
+    const blob = new Blob([JSON.stringify(teachers, null, 2)], {
+      type: "application/json",
+    })
+
+    return new File([blob], "untisPlanner.json", {
+      type: "application/json",
+    })
+  } else return null
+}
```

Comment:
I refactored this to a switch for easier readability

Reply:
okay